dev-python/shiboken/shiboken-9999.ebuild

```diff
@@ -1,38 +1,41 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=6
 
-PYTHON_COMPAT=( python{2_7,3_4,3_5} )
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
 
 inherit cmake-utils python-r1 git-r3
 
 DESCRIPTION="A tool for creating Python bindings for C++ libraries"
-HOMEPAGE="https://wiki.qt.io/Pyside"
+HOMEPAGE="https://wiki.qt.io/PySide2"
 EGIT_REPO_URI=(
-	"git://code.qt.io/pyside/${PN}.git"
-	"https://code.qt.io/git/pyside/${PN}.git"
+	"git://code.qt.io/pyside/shiboken.git"
+	"https://code.qt.io/git/pyside/shiboken.git"
 )
 
 LICENSE="LGPL-2.1"
-SLOT="0"
+SLOT="2"
 KEYWORDS=""
 IUSE="test"
-
 REQUIRED_USE="${PYTHON_REQUIRED_USE}"
 
+# Minimum Qt version required, While PySide2 requires specific Qt 5 releases,
+# Shiboken2 purports to support all Qt 5 releases.
+QT_PV="5:5="
```

Comment:
> I might just be misinterpreting that statement, but shiboken does seem to be breaking privacy encapsulation in certain edge cases.

I think that comment just means that shiboken is not able to parse all C++ constructs and chokes on some code in the private section of that class. This does not imply that shiboken/pyside rely on the private ABI.

> Curiously, the Qt ABI Tracker shows a number of ABI symbol changes across even minor Qt versions

Only **removal** of **public** symbols breaks ABI.

> More relevantly, minor Qt versions sometimes introduce entirely new functionality. Qt 5.5.1, for example, added the new Qt5WaylandClient component.

Adding stuff doesn't break the ABI. Moreover, that module was not considered stable yet at that point in time.

> creating Python bindings for new functionality added between minor Qt versions should still require rebuilding shiboken and pyside.

Correct. In practice, it has never been an issue with PyQt. So I would not introduce a subslot operator for this reason alone.